zero/gyrokinetic_maxwellian_moments.c

```diff
@@ -0,0 +1,233 @@
+#include <math.h>
+
+#include <gkyl_alloc.h>
+#include <gkyl_array.h>
+#include <gkyl_array_ops.h>
+#include <gkyl_array_ops_priv.h>
+#include <gkyl_dg_bin_ops.h>
+#include <gkyl_dg_calc_sr_vars.h>
+#include <gkyl_dg_updater_moment_gyrokinetic.h>
+#include <gkyl_gyrokinetic_maxwellian_moments.h>
+#include <gkyl_gyrokinetic_maxwellian_moments_priv.h>
+
+struct gkyl_gyrokinetic_maxwellian_moments*
+gkyl_gyrokinetic_maxwellian_moments_inew(const struct gkyl_gyrokinetic_maxwellian_moments_inp *inp)
+{
+  gkyl_gyrokinetic_maxwellian_moments *up = gkyl_malloc(sizeof(*up));
+
+  up->conf_basis = *inp->conf_basis;
+  up->phase_basis = *inp->phase_basis;
+  up->num_conf_basis = inp->conf_basis->num_basis;
+  // Determine factor to divide out of temperature computation
+  // If 1x1v, up->vdim = 1, otherwise up->vdim = 3
+  if (up->phase_basis.ndim - up->conf_basis.ndim == 1) {
+    up->vdim = up->phase_basis.ndim - up->conf_basis.ndim;
+  }
+  else {
+    up->vdim = 3;
+  }
+  up->gk_geom = gkyl_gk_geometry_acquire(inp->gk_geom);
+  up->divide_jacobgeo = inp->divide_jacobgeo;
+
+  long conf_range_ncells = inp->conf_range->volume;
+  long conf_range_ext_ncells = inp->conf_range_ext->volume;
+
+  if (inp->use_gpu) {
```

Comment:
@JunoRavin no strong preference on this one, but should we tell `gyrokinetic_maxwellian_moments` what kind of moments we are going to request, so that we only allocate the temporary arrays we need, instead of all of these?
Maybe not since it'll be kind a pain to go through and change it now, and we are not starving for memory AFAICT